src/core/ext/filters/client_channel/resolver/google_c2p/google_c2p_resolver.cc

```diff
@@ -256,9 +256,14 @@ void GoogleCloud2ProdResolver::StartXdsResolver() {
       {"xds_servers", xds_server},
       {"authorities",
        Json::FromObject({
-           {kC2PAuthority, Json::FromObject({
-                               {"xds_servers", std::move(xds_server)},
-                           })},
+           {kC2PAuthority,
+            Json::FromObject({
+                {"xds_servers", std::move(xds_server)},
+                {"client_listener_resource_name_template",
+                 Json::FromString(
+                     "xdstp://traffic-director-c2p.xds.googleapis.com/"
```

Comment:
Suggest writing this as `absl::StrCat("xdstp://", kC2PAuthority, "/envoy.config.listener.v3.Listener/%s")`.

Reply:
done